background-processor/src/lib.rs

```diff
@@ -0,0 +1,292 @@
+use lightning::chain;
+use lightning::chain::chaininterface::{BroadcasterInterface, FeeEstimator};
+use lightning::chain::keysinterface::{ChannelKeys, KeysInterface};
+use lightning::ln::channelmanager;
+use lightning::ln::channelmanager::ChannelManager;
+use lightning::{log_internal, log_error, log_trace};
+use lightning::util::logger::Logger;
+use lightning::util::ser::Writeable;
+use std::sync::Arc;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::thread;
+use std::time::{Duration, Instant};
+
+/// BackgroundProcesser runs a background thread that monitors whether the ChannelManager it's given
+/// on start has new updates to persist. If it does, then it writes said ChannelManager to
+/// disk/backups.
+/// If ChannelManager persistence fails and the persisted manager becomes out-of-date, then there is
+/// a risk of channels force-closing on startup when the manager realizes it's outdated. However, as
+/// long as `ChannelMonitor` backups are sound, no funds besides those used for unilateral chain
+/// closure fees are at risk.
+pub struct BackgroundProcessor {
```

Comment:
Could you update this documentation so it phrased less specifically about channel manager and more generally about other background tasks. It's fine to enumerate everything after a general description and add any caveats, of course.